[gavinsoh] iP

gavinsoh opened the pull request:
## GavinChatBot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

> "The only way to do great work is to love what you do." – Steve Jobs


GavinChatBot frees your mind of having to remember things you need to do. It's,

- text-based :smiley_cat:

- easy to learn :smiley_cat:

- ~~FAST~~ SUPER FAST to use :smiley_cat:

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
And it is **FREE!**

Features:

- [ ] Managing tasks 🙉 
- [ ] Managing deadlines (coming soon) 🙈 
- [ ] Reminders (coming soon) 🙈 

![Screenshot of a comment on a GitHub issue showing an image, added in the Markdown, of an Octocat smiling and raising a tentacle.](https://myoctocat.com/assets/images/base-octocat.svg)

```java
public class Duke {
    public static void main(String[] args) {
        System.out.println("Hello from Duke!");
    }

To compile the program, use the `gcc -o myprogram myprogram.c` command.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

gavinsoh committed:
Level-0

gavinsoh committed:
Level-1

gavinsoh committed:
Level-2

gavinsoh committed:
Level-3

gavinsoh committed:
Level-4

gavinsoh committed:
Level-5

gavinsoh committed:
Level-5

gavinsoh committed:
Level-5

gavinsoh committed:
Level-6

gavinsoh committed:
A-TextUiTesting

gavinsoh committed:
A-TextUiTesting

gavinsoh committed:
Merge branch 'master' of github.com:gavinsoh/ip

gavinsoh committed:
Complete Level 7: Save tasks to file

gavinsoh committed:
Complete Level 8: Dates and Times

gavinsoh committed:
Merge branch 'branch-Level-8'

git tag Level-8

gavinsoh committed:
A-MoreOOP

gavinsoh committed:
A-Packages

gavinsoh committed:
Merge branch 'add-gradle-support'

gavinsoh committed:
A-Gradle

gavinsoh committed:
A-JavaDoc

gavinsoh committed:
A-CodingStandard

gavinsoh committed:
Level-9

gavinsoh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/gavinchatbot/task/Task.java
#	src/main/java/gavinchatbot/util/Ui.java

gavinsoh committed:
A-CheckStyle

gavinsoh committed:
Level-10

gavinsoh committed:
Level-10

justsparsh commented on src/main/java/gavinchatbot/util/Ui.java:
```diff
@@ -0,0 +1,165 @@
+     * Returns a welcome message to the user.
+     */
+    public String showWelcome() {
+        String horizontalLine = "_________________________________________________________________________________\n";
```
Perhaps the variable horizontalLine can be extracted as a constant:

``` ruby
String HORIZONTAL_LINE = "_________________________________________________________________________________\n";
```

gavinsoh replied:
That is a great suggestion! I will make sure to implement this change.

justsparsh commented on src/main/java/gavinchatbot/util/Parser.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Represents a parser that interprets user input and returns the appropriate command.
+ */
+public class Parser {
```
I like how you have JavaDocs for all classes

gavinsoh replied:
Thank you Sparsh

justsparsh commented on src/main/java/gavinchatbot/task/ToDos.java:
```diff
@@ -0,0 +1,37 @@
+ * Represents a task that is a to-do.
+ * Inherits from the Task class.
+ */
+public class ToDos extends Task {
```
Perhaps this class would be better named as `ToDo`

gavinsoh replied:
Thank you Sparsh, I will take note!

kahsuann commented on src/main/java/gavinchatbot/GavinChatBot.java:
```diff
@@ -0,0 +1,53 @@
+     * @param input The user's input.
+     * @return The chatbot's response.
+     */
+    public String getResponse(String input) {
+        try {
+            Command command = new Parser().parse(input);
+            return command.execute(tasks, ui, storage);
+        } catch (GavinException | IOException e) {
+            return ui.showError("Error: " + e.getMessage());
+        }
+    }
```
I like that you use verbs for method names / actions!

gavinsoh replied:
Thank you!

kahsuann commented on src/main/java/gavinchatbot/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,58 @@
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws GavinException, IOException {
+        Task task = new Deadline(description, by.toString());
+        tasks.addTask(task);
+        storage.save(tasks.getTasks());
+        return ui.showAddedTask(task, tasks.size());
```
It's great that the names explain the code well to the reader.

gavinsoh replied:
Thank you.

kahsuann reviewed (commented):
LGTM! Names are not misleading and are informative (not too long not too short)

gavinsoh committed:
A-Assertions: Add assertions for task index and task description validation

gavinsoh committed:
A-CodeQuality: Refactor task handling and improve code quality

gavinsoh committed:
A-Streams: Refactor task list to use Java Streams

gavinsoh committed:
Merge pull request #1 from gavinsoh/branch-A-Assertions

A-Assertions: Add assertions for task index and task description validation

gavinsoh committed:
A-CodeQuality: Refactor task handling logic

- Extracted task index validation logic to improve code readability and reusability.
- Simplified repetitive task creation methods by using a helper method.
- Improved naming conventions and ensured consistency.

gavinsoh committed:
Merge branch 'master' into branch-A-Streams

A-Streams: Merge master into branch-A-Streams

- Merged the latest updates from the master branch into branch-A-Streams.
- Ensured consistency by syncing the changes before merging Streams-related features.

gavinsoh committed:
Merge branch 'master' into branch-A-CodeQuality

gavinsoh committed:
Merge pull request #2 from gavinsoh/branch-A-CodeQuality

A-CodeQuality: Refactor task handling logic

- Extracted task index validation logic to improve code readability and reusability.
- Simplified repetitive task creation methods by using a helper method.
- Improved naming conventions and ensured consistency.

gavinsoh committed:
Merge branch 'master' into branch-A-Streams

A-Streams: Merge master into branch-A-Streams

- Synced branch-A-Streams with the latest changes from the master branch.
- Ensured compatibility between the latest updates and the Streams-based implementation.
- Resolved any potential merge conflicts and maintained stream operations for task filtering.

gavinsoh committed:
Merge pull request #3 from gavinsoh/branch-A-Streams

A-Streams: Use Java Streams for task filtering

- Replaced traditional loops with Java Streams for filtering tasks marked as done.
- Improved the performance and readability of the code by leveraging Stream APIs.

gavinsoh committed:
Merge branch 'branch-A-CodeQuality' of github.com:gavinsoh/ip into branch-A-CodeQuality

gavinsoh committed:
Fix CheckStyle violation in TaskList: Line length exceeds 120 characters

- Refactored a line in the TaskList file to meet the 120-character limit for better readability and adherence to CheckStyle guidelines

gavinsoh committed:
fix(branches): Rectify incorrect branch pulling

- Accidentally messed up the branches during a pull operation.
- Now attempting to rectify and bring branches back to their correct state.
- No functional changes yet, just fixing the repository structure.

gavinsoh committed:
Merge branch 'master' of github.com:gavinsoh/ip

Merge master into branch-A-Streams to sync changes

- Pulled the latest changes from the master branch to ensure branch-A-Streams is up-to-date.
- This merge is necessary to resolve the differences between the local changes and the updates made in the master branch.
- This will prevent conflicts and ensure that the latest features and fixes from the master branch are included in the current stream of work.

gavinsoh committed:
Fix CheckStyle violation in TaskList: Line length exceeds 120 characters

- Refactored a line in the TaskList file to meet the 120-character limit for better readability and adherence to CheckStyle guidelines

gavinsoh committed:
BCD-Extension: Add TagCommand for tagging tasks with custom tags

gavinsoh committed:
fix(UI): Ensure welcome message is displayed on startup

gavinsoh committed:
Merge branch 'branch-BCD-Extension'

Merge branch 'branch-BCD-Extension' into master

- This merge brings in the TagCommand feature for tagging tasks with custom tags.
- Includes the latest changes to ensure task management is more efficient with tagging functionality.

gavinsoh committed:
A-Personality: Refine chatbot response phrases for user commands

Tweaked the phrases used by the chatbot when responding to user
commands to improve clarity and friendliness.

gavinsoh committed:
Added Ui.png for graphical user interface layout

gavinsoh committed:
A-UserGuide: Update user guide with detailed feature explanations

- Added sections for adding tasks, including ToDo, Deadline, and Event.
- Updated instructions for listing tasks and finding tasks by keywords.
- Included features for marking, unmarking, deleting, and tagging tasks.

gavinsoh committed:
Update comments for clarity and accuracy

Refine documentation and inline comments to improve code readability.

gavinsoh committed:
Update comments for clarity and accuracy

Refine documentation and inline comments to improve code readability.

gavinsoh committed:
Update comments for clarity and accuracy

Refine documentation and inline comments to improve code readability.

gavinsoh committed:
Update comments for clarity and accuracy

Refine documentation and inline comments to improve code readability.

gavinsoh committed:
Update comments for clarity and accuracy

Refine documentation and inline comments to improve code readability.

gavinsoh committed:
Update JUnit tests for enhanced coverage

Refine and extend JUnit tests to ensure reliable function behavior.
